### ISSUE TITLE
`secure_spoke_virtual_network_resource_ids` doesn't work without additional config outside the module

### ISSUE THREAD

**Issue by tlfzhylj** [Azure/terraform-azurerm-caf-enterprise-scale#632]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Versions

**terraform**: 1.3.8

**azure provider**:  3.44.1

**module**: 3.1.2

#### Describe the bug

The option for `secure_spoke_virtual_network_resource_ids` does not work if the steps described in this issue is not followed:
* https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/issues/491#issuecomment-1353390545
* https://github.com/nthewara/tf-secured-virtualhub

This can give a false expectation from people trying implement this, and believe the vnets are secured.

Also, I did not find any information about this in the docs, but I found it when searching through closed issues. The post in the issue above should be implemented in the docs.

**Comment by matt-FFFFFF** [comment]
Hi @tlfzhylj 

Have you seen the accompanying [Azure/terraform-azurerm-lz-vending](https://github.com/Azure/terraform-azurerm-lz-vending) module? This has just released the capability to configure these settings for spoke vnets.

You can use this module with existing subscriptions as well.

Therefore, what do we need to be able to set in this module to correctly configure the Azure Firewall?

**Comment by tlfzhylj** [comment]
When I configure the following outside of the module, I am able to secure internet traffic. If this is not set, the vnets configured in `secure_spoke_virtual_network_resource_ids` internet traffic will not be secured.

```hcl
resource "azurerm_virtual_hub_route_table_route" "secure_traffic" {
  name           = "all_traffic" # Name needs to be 'all_traffic' to reflect the configuration in Azure Portal. Ref. https://learn.microsoft.com/en-us/azure/firewall-manager/secure-cloud-network-powershell#deploy-azure-firewall-and-configure-custom-routing
  route_table_id = "${values(module.caf-enterprise-scale.azurerm_virtual_hub["virtual_wan"])[0].id}/hubRouteTables/defaultRouteTable"

  destinations_type = "CIDR"
  destinations      = ["0.0.0.0/0", "10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16"]

  next_hop_type = "ResourceId"
  next_hop      = values(module.caf-enterprise-scale.azurerm_firewall["virtual_wan"])[0].id

  provider = azurerm.connectivity
}
```

As shown in [this repo](https://github.com/nthewara/tf-secured-virtualhub), you can also have other configurations, based on your needs.

**Comment by tlfzhylj** [comment]
Another question, which is related:
Private traffic is set to unsecured when peering a spoke with the hub. Is this by design.

Is this by design?

Here's a screenshot to show you what I mean (taken from a Stackoverflow post):
![](https://i.stack.imgur.com/fDHwU.png)

**Comment by tlfzhylj** [comment]
I tried the ALZ vending module now, and see how I can fix my last question regarding securing private traffic.

Propogated route table needs to be set to none like this under the resource `azurerm_virtual_hub_connection`.

```hcl
propagated_route_table {
      route_table_ids = ["${values(module.caf-enterprise-scale.azurerm_virtual_hub["virtual_wan"])[0].id}/hubRouteTables/noneRouteTable"]
      labels = ["none"]
    }
```

I think I will configure peering outside the module.

**Comment by tlfzhylj** [comment]
Anyhow, I still think the bug this issue originally was about needs to either be documented or fixed in code. I may be wrong, and something I don't understand like I tried to explain here: https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/issues/632#issuecomment-1451547661

**Comment by matt-FFFFFF** [comment]
> I tried the ALZ vending module now, and see how I can fix my last question regarding securing private traffic.
> 
> 
> 
> Propogated route table needs to be set to none like this under the resource `azurerm_virtual_hub_connection`.
> 
> 
> 
> ```hcl
> 
> propagated_route_table {
> 
>       route_table_ids = ["${values(module.caf-enterprise-scale.azurerm_virtual_hub["virtual_wan"])[0].id}/hubRouteTables/noneRouteTable"]
> 
>       labels = ["none"]
> 
>     }
> 
> ```
> 
> 
> 
> I think I will configure peering outside the module.

I recommend you use lz-vending to configure the vhub connection. You can then set this route table association there.

The only thing that needs to be done is to ensure that the route tables exist in the vhub.

**Comment by matt-FFFFFF** [comment]
> Anyhow, I still think the bug this issue originally was about needs to either be documented or fixed in code. I may be wrong, and something I don't understand like I tried to explain here: https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/issues/632#issuecomment-1451547661

I agree this should be better documented, or placed into the module. Thanks for the feedback!

**Comment by matt-FFFFFF** [comment]
Hi @MCrank 

You're right. Anything more complex should be managed outside this module.

The lz-vending module can work with existing subscriptions and supports much more complex vwan connection configurations.

**Comment by qaiserali** [comment]
Can we enable this feature for the traditional hub vnet, as we are not using Virtual WAN network resources?
